src/rpc/registry/actors/verified_reg.rs

```diff
@@ -0,0 +1,251 @@
+            ]
+        );
+    }};
+}
+
+macro_rules! register_universal_receiver_hook {
+    ($registry:expr, $cid:expr, $version:tt) => {
+        paste! {
+            {
+                use fil_actor_verifreg_state::[<$version>]::Method;
+                register_actor_methods!(
+                    $registry,
+                    $cid,
+                    [(
+                        Method::UniversalReceiverHook,
+                        fvm_actor_utils::receiver::UniversalReceiverParams
+                    ),]
+                );
+            }
+        }
+    };
+}
+
+fn register_verified_reg_v8(registry: &mut MethodRegistry, cid: Cid) {
+    register_core_methods!(registry, cid, v8);
+    register_v8_unique_methods!(registry, cid);
+}
+
+fn register_verified_reg_v9(registry: &mut MethodRegistry, cid: Cid) {
+    register_core_methods!(registry, cid, v9);
+    register_allocation_methods!(registry, cid, v9);
+    register_universal_receiver_hook!(registry, cid, v9);
+}
+
+fn register_verified_reg_v10(registry: &mut MethodRegistry, cid: Cid) {
+    register_core_methods!(registry, cid, v10);
+    register_allocation_methods!(registry, cid, fil_actor_verifreg_state::v10, no_claim);
+    register_exported_methods!(registry, cid, fil_actor_verifreg_state::v10);
+    register_universal_receiver_hook!(registry, cid, v10);
+}
+
+fn register_verified_reg_v11(registry: &mut MethodRegistry, cid: Cid) {
+    register_core_methods!(registry, cid, fil_actor_verifreg_state::v11);
+    register_allocation_methods!(registry, cid, fil_actor_verifreg_state::v11, no_claim);
+    register_exported_methods!(registry, cid, fil_actor_verifreg_state::v11);
+    register_universal_receiver_hook!(registry, cid, v11);
+}
+
+macro_rules! register_verified_reg_v12_plus {
+    ($registry:expr, $cid:expr, $state_version:path, $version:tt) => {{
+        register_core_methods!($registry, $cid, $state_version);
+        register_allocation_methods!($registry, $cid, $state_version);
+        register_exported_methods!($registry, $cid, $state_version);
+        register_universal_receiver_hook!($registry, $cid, $version);
+    }};
+}
+
+pub(crate) fn register_actor_methods(registry: &mut MethodRegistry, cid: Cid, version: u64) {
+    match version {
+        8 => register_verified_reg_v8(registry, cid),
+        9 => register_verified_reg_v9(registry, cid),
+        10 => register_verified_reg_v10(registry, cid),
+        11 => register_verified_reg_v11(registry, cid),
+        12 => register_verified_reg_v12_plus!(registry, cid, fil_actor_verifreg_state::v12, v12),
+        13 => register_verified_reg_v12_plus!(registry, cid, fil_actor_verifreg_state::v13, v13),
+        14 => register_verified_reg_v12_plus!(registry, cid, fil_actor_verifreg_state::v14, v14),
+        15 => register_verified_reg_v12_plus!(registry, cid, fil_actor_verifreg_state::v15, v15),
+        16 => register_verified_reg_v12_plus!(registry, cid, fil_actor_verifreg_state::v16, v16),
+        _ => {}
```

Comment:
it'd be better to actually report some error in this case

Reply:
Got it, I’ll keep this in mind.

I’m planning to open a PR once we’ve implemented all the actor methods. In that PR, we can review the overall design and suggest changes like renaming files or doing cleanups. I’ll introduce this change in that PR as well.